src/posix/platform/infra_if.cpp

```diff
@@ -258,7 +258,7 @@ bool InfraNetif::IsRunning(void) const
     VerifyOrDie(sock != -1, OT_EXIT_ERROR_ERRNO);
 
     memset(&ifReq, 0, sizeof(ifReq));
-    strncpy(ifReq.ifr_name, mInfraIfName, sizeof(ifReq.ifr_name));
+    strncpy(ifReq.ifr_name, mInfraIfName, sizeof(ifReq.ifr_name) - 1);
```

Comment:
use strcpy here as well? We may add a static_assert here to guarantee sizeof(ifReq.ifr_name) == sizeof(mInfraIfName).